library/alloc/src/vec/mod.rs

```diff
@@ -1001,6 +1001,22 @@ impl<T, A: Allocator> Vec<T, A> {
         }
     }
 
+    /// Converts the vector into [`Box<T>`][owned slice], assuming that
+    /// `self.len() == self.capacity()`.
+    ///
+    /// # Safety
+    /// It's undefined behavior to call this function if `self.capacity() > self.len()`.
+    #[cfg(not(no_global_oom_handling))]
+    pub(crate) unsafe fn into_exact_size_boxed_slice(self) -> Box<[T], A> {
+        debug_assert_eq!(self.len(), self.capacity());
```

Comment:
This assert could break if the RawVec implementation gets changed to (again) return the excess capacity. So this should have at least have a comment saying that it depends on the current implementation and it's ok to remove it if the assumptions change.